server/src/main/java/io/crate/planner/operators/HashJoin.java

```diff
@@ -129,11 +121,9 @@ public ExecutionPlan build(PlannerContext plannerContext,
         LogicalPlan leftLogicalPlan = lhs;
         LogicalPlan rightLogicalPlan = rhs;
 
-        boolean tablesSwitched = false;
```

Comment:
no-need for this flag anymore, sides are already flipped when the hash symbols are derived.